src/test/java/com/limechain/sync/fullsync/BlockExecutorTest.java

```diff
@@ -23,68 +25,46 @@
-        header.setParentHash(Hash256.from("0xb0a8d493285c2df73290dfb7e61f870f17b41801197a149ca93654499ea3dafe"));
-        header.setBlockNumber(BigInteger.valueOf(1));
-        header.setStateRoot(Hash256.from("0xfabb0c6e92d29e8bb2167f3c6fb0ddeb956a4278a3cf853661af74a076fc9cb7"));
-        header.setExtrinsicsRoot(Hash256.from("0xa35fb7f7616f5c979d48222b3d2fa7cb2331ef73954726714d91ca945cc34fd8"));
-        header.setDigest(
-                ScaleUtils.Decode.decodeList(scaleEncodedDigest, new HeaderDigestReader()).toArray(HeaderDigest[]::new));
+        // Build the genesis state trie
+        ChainSpec chainSpec = ChainSpec.newFromJSON("genesis/ksmcc3.json");
+        var genesisStorage = chainSpec.getGenesis().getTop();
+        var runtimeByteCode = genesisStorage.get(ByteString.copyFrom(":code".getBytes())).toByteArray();
+        var genesisTrie = TrieStructureFactory.buildFromKVPs(genesisStorage, StateVersion.V0);
 
-        var exts = ScaleUtils.Decode.decodeList(scaleEncodedBody, ScaleCodecReader::readByteArray);
-        assertEquals(2, exts.size());
-        BlockBody body = new BlockBody(exts.stream().map(Extrinsics::new).toList());
-        Block block = new Block(header, body);
+        // Define dependencies for the execution context
+        KVRepository<String, Object> db = new InMemoryDB(); // initialize a DB
 
-        ChainSpec chainSpec = ChainSpec.newFromJSON("genesis/ksmcc3.json");
-        var top = chainSpec.getGenesis().getTop();
+        TrieStorage trieStorage = new TrieStorage(db); // inject it into a trie storage
+        trieStorage.insertTrieStorage(genesisTrie, StateVersion.V0); // populate with the genesis trie
 
-        var runtimeByteCode = top.get(ByteString.copyFrom(":code".getBytes())).toByteArray();
+        final byte[] genesisStateRoot = StringUtils.hexToBytes("b0006203c3a6e6bd2c6a17b1d4ae8ca49a31da0f4579da950b127774b44aef6b");
+        BlockTrieAccessor trieAccessor = new BlockTrieAccessor(trieStorage, genesisStateRoot); // instantiate a block trie accessor with the trieStorage specified for the genesis block
 
-//        // Dumps the .wasm file for manual conversion to .wat
-//         FileUtils.writeByteArrayToFile(new File("/tmp/kusama1.wasm"), runtimeByteCode);
+        // Package all dependencies into the expected configuration for the runtime builder
+        var cfg = new RuntimeBuilder.Config(trieAccessor, null, null, false);
 
-        Runtime genesisRuntime = new RuntimeBuilder().buildRuntime(runtimeByteCode);
-        byte[] encodedUnsealedHeader =
-                ScaleUtils.Encode.encode(BlockHeaderScaleWriter.getInstance()::writeUnsealed, block.getHeader());
+        Runtime genesisRuntime = new RuntimeBuilder().buildRuntime(runtimeByteCode, cfg);
 
-        var encodedBody = ScaleUtils.Encode.encode(BlockBodyWriter.getInstance(), body);
+        // Now, prepare the arguments for the actual runtime calls
+        // TODO: Encapsulate these more properly
```

Comment:
The test is not yet finished (in the sense of being quality test code), but it otherwise does what's intended (in this sense, it's finished). 

Rushing out due to lack of time.

Reply:
Not sure what the question is but you can modify/delete this test as you see fit.

Reply:
This test is the pinnacle of what we're striving for, so I wouldn't delete it ever. 😄 
It is just not "beautiful" since it's still in its sketch form.
Delegated to the follow-up PR.

Reply:
Fixed in [e222d87](https://github.com/LimeChain/Fruzhin/pull/464/commits/e222d870a46afcb797ef1ab9b9517f2b8c8bd6dc)